src/com/amazon/ion/impl/IonReaderBinarySystemX.java

```diff
@@ -296,40 +295,35 @@ public double doubleValue()
         return _v.getDouble();
     }
 
-    public int intValue()
+    private void checkIsIntApplicableType()
```

Comment:
This method is duplicated in both binary/text readers. Not clear for me where I should move it. Could you please suggest some util class?

Reply:
the binary/text readers don't have a common ancestor so there is good place to move it without a considerable amount of refactoring. Adding the method already removes some duplication so I think this is good for now